packages/ado-extension/task.json

```
@@ -29,61 +29,47 @@
             "type": "pickList",
-            "label": "Hosting Mode",
+            "label": "Hosting mode",
             "required": true,
             "options": {
-                "staticSite": "Static Site (this task runs a localhost webserver for your site)",
-                "dynamicSite": "Dynamic Site (you host your site separately)"
+                "staticSite": "Static site (HTML files in a local directory)",
+                "dynamicSite": "URL (localhost, staging URL, or production URL)"
             },
-            "helpMarkDown": "Note: not required in YAML configuration, only on classic pipelines UI. Your site must be served (hosted) before it can be scanned. In _Static Site_ mode, this task will run a localhost webserver serving your site directory and scan that. In _Dynamic Site_ mode, you must host your site yourself separately and specify a URL to scan. This can be either a localhost server you run in an earlier pipeline step, or a remote URL (for example, a staging environment)."
+            "helpMarkDown": "Note: not required in YAML configuration, only on classic pipelines UI. Your site must be served (hosted) before it can be scanned. In _Static site_ mode, this task will run a localhost webserver serving your site directory and scan that. In _URL_ mode, you must host your site yourself separately and specify a URL to scan. This can be either a localhost server you run in an earlier pipeline step, or a remote URL (for example, a staging environment).\n\nWhen switching between hosting modes in the classic pipeline, remove values for the previous hosting mode to prevent caching."
         },
         {
             "name": "staticSiteDir",
             "type": "string",
-            "label": "Static Site Directory",
+            "label": "Hosting mode: site directory",
             "required": true,
-            "helpMarkDown": "Folder containing website content.",
+            "helpMarkDown": "Directory containing the website content to be scanned.",
             "visibleRule": "hostingMode = staticSite"
         },
         {
             "name": "staticSiteUrlRelativePath",
             "type": "string",
-            "label": "Static Site URL Relative Path",
+            "label": "Hosting mode: URL relative path",
             "required": false,
             "defaultValue": "/",
-            "helpMarkDown": "Relative path to directory used to construct base scan url. e.g. / on Ubuntu and // on Windows.",
+            "helpMarkDown": "Relative path to directory used to construct base scan url. For example, use `/` on Ubuntu and `//` on Windows.",
             "visibleRule": "hostingMode = staticSite"
         },
         {
             "name": "staticSitePort",
             "type": "int",
-            "label": "Static Site Port",
+            "label": "Hosting mode: port",
             "required": false,
             "helpMarkDown": "The preferred local website TCP port to use when scanning local website content.",
```

Honestly, this one is confusing to me too. The wording makes it seem like the user should know which port we need to scan but in `staticSite` mode, the extension sets up the server. If this setting exists specifically to ensure that we don't serve on a port already in use in the pipeline, it might be good to call that out. Something like
```suggestion
            "helpMarkDown": "The preferred local website TCP port to use when serving local website content. Defaults to ${port}. Set this to avoid conflicts if ${port} is already in use in this pipeline.",
```

---

Totally agree! I believe that there is not a default port and one is randomly selected at each run. Perhaps, we could try....

```suggestion
            "helpMarkDown": "The preferred local website TCP port to use when serving local website content. If unspecified, a port will be set for you. If the pipeline serves additional processes, set the port to avoid conflicts. If using baselining, set the port to make sure future scans will match the baseline file.",
```